pkg/daemon/agent/flexvolume/server.go

```diff
@@ -106,30 +109,47 @@ func (s *FlexvolumeServer) Start() error {
 }
 
 // Stop the unix domain socket server and deletes the socket file
-func (s *FlexvolumeServer) Stop() {
-	if s.listener != nil {
-		logger.Info("Stopping unix socket rpc server.")
-		if err := s.listener.Close(); err != nil {
+func (s *FlexvolumeServer) StopAll() {
+	logger.Infof("Stopping %d unix socket rpc server(s).", len(s.listeners))
+	for unixSocketFile, listener := range s.listeners {
+		if err := listener.Close(); err != nil {
 			logger.Errorf("Failed to stop unix socket rpc server: %+v", err)
 		}
+
+		// closing the listener should remove the unix socket file. But lets try it remove it just in case.
+		if _, err := os.Stat(unixSocketFile); !os.IsNotExist(err) {
+			logger.Infof("Deleting unix domain socket file %s.", unixSocketFile)
+			os.Remove(unixSocketFile)
+		}
 	}
-	// closing the listener should remove the unix socket file. But lets try it remove it just in case.
-	unixSocketFile := path.Join(flexVolumeDriverDir, path.Join(UnixSocketName)) // /flextmnt/rook.io~rook/.rook.sock
-	if _, err := os.Stat(unixSocketFile); !os.IsNotExist(err) {
-		logger.Info("Deleting unix domain socket file.")
-		os.Remove(unixSocketFile)
-	}
+	s.listeners = make(map[string]net.Listener)
+}
 
+func RookDriverName(context *clusterd.Context) (string, error) {
+	kubeVersion, err := k8sutil.GetK8SVersion(context.Clientset)
+	if err != nil {
+		return "", fmt.Errorf("Error getting server version: %v", err)
+	}
+	if kubeVersion.AtLeast(version.MustParseSemantic(serverVersionV180)) {
```

Comment:
i'm forgetting what k8s 1.8+ has to do with being able to use a namespaced driver name, please remind me

Reply:
1.7 doesn't appear to support multiple drivers under the same rook.io provider. I'll add a comment here